mpy-cross/mpconfigport.h

```diff
@@ -90,76 +76,10 @@
 #define MICROPY_PY_IO               (0)
 #define MICROPY_PY_SYS              (0)
 
-// type definitions for the specific machine
-
-#ifdef __LP64__
-typedef long mp_int_t; // must be pointer size
-typedef unsigned long mp_uint_t; // must be pointer size
-#elif defined(__MINGW32__) && defined(_WIN64)
-#include <stdint.h>
-typedef __int64 mp_int_t;
-typedef unsigned __int64 mp_uint_t;
-#elif defined(_MSC_VER) && defined(_WIN64)
-typedef __int64 mp_int_t;
-typedef unsigned __int64 mp_uint_t;
-#else
-// These are definitions for machines where sizeof(int) == sizeof(void*),
-// regardless for actual size.
-typedef int mp_int_t; // must be pointer size
-typedef unsigned int mp_uint_t; // must be pointer size
-#endif
-
-// Cannot include <sys/types.h>, as it may lead to symbol name clashes
-#if _FILE_OFFSET_BITS == 64 && !defined(__LP64__)
-typedef long long mp_off_t;
-#else
-typedef long mp_off_t;
-#endif
-
+// Because code never executes and error handling is done by mpy-cross itself,
+// there is no need to be able to print.
+#define MICROPY_USE_INTERNAL_PRINTF (0)
 #define MP_PLAT_PRINT_STRN(str, len) (void)0
 
-// We need to provide a declaration/definition of alloca()
-#if defined(__FreeBSD__) || defined(__NetBSD__)
-#include <stdlib.h>
-#elif defined(_WIN32)
-#include <malloc.h>
-#else
-#include <alloca.h>
-#endif
-
-#include <stdint.h>
-
-// MSVC specifics - see windows/mpconfigport.h for explanation
-#ifdef _MSC_VER
-
-#define MP_ENDIANNESS_LITTLE        (1)
-#define NORETURN                    __declspec(noreturn)
-#define MP_NOINLINE                 __declspec(noinline)
-#define MP_ALWAYSINLINE             __forceinline
-#define MP_LIKELY(x)                (x)
-#define MP_UNLIKELY(x)              (x)
-#define MICROPY_PORT_CONSTANTS      { MP_ROM_QSTR(MP_QSTR_dummy), MP_ROM_PTR(NULL) }
-#ifdef _WIN64
-#define MP_SSIZE_MAX                _I64_MAX
-#else
-#define MP_SSIZE_MAX                _I32_MAX
-#endif
-#define MICROPY_MAKE_POINTER_CALLABLE(p) ((void *)(p)) // Avoid compiler warning about different const qualifiers
-#define restrict
-#define inline                      __inline
-#define alignof(t)                  __alignof(t)
-#undef MICROPY_ALLOC_PATH_MAX
-#define MICROPY_ALLOC_PATH_MAX      260
-#define PATH_MAX                    MICROPY_ALLOC_PATH_MAX
-#define S_ISREG(m)                  (((m) & S_IFMT) == S_IFREG)
-#define S_ISDIR(m)                  (((m) & S_IFMT) == S_IFDIR)
-#ifdef _WIN64
-#define SSIZE_MAX                   _I64_MAX
-typedef __int64 ssize_t;
-#else
-#define SSIZE_MAX                   _I32_MAX
-typedef int ssize_t;
-#endif
-typedef mp_off_t off_t;
-
-#endif
+// Specific configuration for compiling mpy-cross as a native binary.
+#include "mpconfigport_main.h"
```

Comment:
I've renamed it to "mpconfigport.h" because making the Windows Visual Studio project correctly generate the `MP_CONFIGFILE` variable is too difficult. (In summary: you can easily add the preprocessor definition, using `%22` for the quotes, but then these quotes get lost passing them down to makeqstrdata.py. You can do further transformations to re-do the quote escaping but it's a total mess and not worth the hassle). But I've kept it using MP_CONFIGFILE for the wasm build to set "mpconfigport_wasm.h".

I think conceptually it also makes sense, we have the "default" build (similar to how the Makefile has no suffix, hence it should be just "mpconfigport.h" too), and the "wasm" build is the special one. And this side-steps the awkwardness of what to call it... "main", "native", etc.